packages/ActionBar/src/hooks/useColumnsArrangement/useColumnsArrangement.js

```diff
@@ -1,18 +1,29 @@
 import React from "react";
 import produce from "immer";
 
-export default function useColumnsArrangement(defaultOrder) {
+export default function useColumnsArrangement(defaultOrder, disabledColumnIds = []) {
   const [order, setOrder] = React.useState(defaultOrder);
   const [hiddenColumnIds, setHiddenColumnIds] = React.useState([]);
 
-  const isColumnHidden = React.useCallback(
-    columnId => {
-      return hiddenColumnIds.indexOf(columnId) > -1;
-    },
-    [hiddenColumnIds]
-  );
+  function canMove({ source, destination }) {
+    if (disabledColumnIds.length === 0) return true;
+
+    const newOrder = [...order];
+    const movedChild = newOrder.splice(source, 1);
+    newOrder.splice(destination, 0, ...movedChild);
+
+    return disabledColumnIds.every(
+      disabledColumnId => defaultOrder.indexOf(disabledColumnId) === newOrder.indexOf(disabledColumnId)
+    );
+  }
+
+  function isColumnHidden(columnId) {
+    return hiddenColumnIds.indexOf(columnId) > -1;
```

Comment:
`nit` 
```js
hiddenColumnIds.includes(columnId)
```

Reply:
I don't know why we don't do it, but I think we should start using more Javascript Sets, this and remove would be super simple if we were using sets instead of arrays. 

`SomeSet.has(columnId) // :P`

Reply:
I think usually the main reason for using an array is the order matters.I can convert `hiddenColumnIds ` to a set for sure!